zebrad/src/components/tracing/component.rs

```diff
@@ -86,7 +86,7 @@ impl Tracing {
         let use_color = config.use_color_stdout();
         let use_color_stderr = config.use_color_stderr();
 
-        let filter = config.filter.unwrap_or_default();
+        let filter = config.filter.clone().unwrap_or_default();
```

Comment:
Could we use `default_log_file()` on line 117:

```
let writer = if let Some(log_file) = config.log_file.as_ref() {
```

Reply:
I don't have enough context to understand why we would want to make this change. Is it optional, or does it fix a bug?

Reply:
It's so that it uses a default log file path when no path is provided in the config but progress bars are enabled.

Reply:
This is fixed during deserialization in e154a5a4a6 using a wrapper type. This is the recommended workaround for a missing post-deserialization feature in serde.